src/proxy/http2/Http2ConnectionState.cc

```diff
@@ -852,9 +852,14 @@ Http2ConnectionState::rcv_goaway_frame(const Http2Frame &frame)
   while (nbytes < frame.header().length) {
     unsigned read_bytes = read_rcv_buffer(buf, sizeof(buf), nbytes, frame);
 
-    if (!http2_parse_goaway(make_iovec(buf, read_bytes), goaway)) {
-      return Http2Error(Http2ErrorClass::HTTP2_ERROR_CLASS_CONNECTION, Http2ErrorCode::HTTP2_ERROR_PROTOCOL_ERROR,
-                        "goaway failed parse");
+    if (nbytes == HTTP2_GOAWAY_LEN) {
```

Comment:
Can't we simply remove the while loop instead? I think the while loop is unnecessary for this frame (it makes sense for SETTINGS frame).

Reply:
Yeah, I think we can, but I'm not sure whether we plan on handling the `Additional Debug Data` part in the future. I'm down for removing the loop all together.